src/WebJobs.Extensions.DurableTask/DurableOrchestrationClient.cs

```diff
@@ -50,7 +50,7 @@ public override HttpResponseMessage CreateCheckStatusResponse(HttpRequestMessage
         }
 
         /// <inheritdoc />
-        public override async Task<string> StartNewAsync(string orchestratorFunctionName, string instanceId, object input)
+        public override async Task<string> StartNewAsync(string orchestratorFunctionName, string instanceId, object input, bool waitUntilOrchestrationStarts = false)
```

Comment:
I wonder if we should set `waitUntilOrchestrationStarts = true` by default. The problem we have now is that the behavior does not match peoples' expectations. If we wait by default, then we will meet people's expectations and I suspect it will match the default behavior for the longer-term fix. For those who don't care about waiting, they can set this to `false` explicitly. Thoughts?

Reply:
Great! I really like this approach. You are right, developers expect this as the default behavior. Thank you!